### PR TITLE
Implement minimal persistence

### PR DESCRIPTION
**Pull request by vldm**
Closes: https://github.com/stegos/stegos/issues/59
Closes: https://github.com/stegos/stegos/issues/57
Closes: https://github.com/stegos/stegos/issues/58
Closes: https://github.com/stegos/stegos/issues/8

Fixed a bug in VRF system: VRF system didn't restart at low connectivity setup.

Minimal persistence implementation. For now node recover after shutdown.
TODO:
- [x] Recover wallet.
- [x] VRF system stop working if after restart cannot connect to other nodes.

### CODE REVIEW

File: blockchain/Cargo.toml [hunk 1]
```diff
@@ -10,12 +10,16 @@ build = "build.rs"
 stegos_keychain = { path = "../keychain" }
 stegos_crypto = { path = "../crypto" }
 stegos_serialization = {path = "../serialization"}
+rocksdb = "0.11.0"
```
[the remark this answers is not in the document]
Comment (vldm) [reply blockchain/Cargo.toml]: Cargo will allow update you, even if you specify minor version, it use SemVer(slightly modifyed) to choose version.

"0.11.0", "^0.11.0", "^0.11", "0.11", are all the same.

[the remark this answers is not in the document]
Comment (vldm) [reply blockchain/Cargo.toml]: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

File: blockchain/src/storage.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        let db = DB::open_default(path).expect("couldn't open database");
+        Self { db }
+    }
+
+    pub fn insert(&self, len: u64, block: Block) {
+        let data = block.into_buffer().expect("couldn't serialize block.");
+
+        let mut batch = WriteBatch::default();
+        // writebatch put fails if size exceeded u32::max, which is not our case.
+        batch.put(&Self::u64_to_bytes(len), &data).unwrap();
```
[the remark this answers is not in the document]
Comment (vldm) [reply blockchain/src/storage.rs]: Prefixes is not the best choose for iterators.
We can use different column fameles instead.  (https://github.com/facebook/rocksdb/wiki/Column-Families)

File: blockchain/src/storage.rs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+use rocksdb::{IteratorMode, WriteBatch, DB};
+use stegos_serialization::traits::ProtoConvert;
+use tempdir::TempDir;
+
+use std::ops::Deref;
+use std::path::Path;
+
+use super::block::Block;
+
+pub struct ListDB {
```
[the remark this answers is not in the document]
Comment (vldm) [reply blockchain/src/storage.rs]: @rtsisyk Database that is only abble to save "list"

File: blockchain/src/storage.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vldm) [reply blockchain/src/storage.rs]: Okay, i will modify it

File: node/src/lib.rs [hunk 4]
```diff
@@ -379,6 +386,82 @@ where
         Ok(())
     }
 
+    fn recover_state(&mut self) {
```
[the remark this answers is not in the document]
Comment (vldm) [reply node/src/lib.rs]: It's not fully 1to1 rewriting of `on_key_block_registred`, not all state are recoverable from keyblock.

We can keep refactor our code to save "latest keyblock" in blockchain rather then current (leader, facilitator, validators), but i think that this modification is good after we go into the second phase of our persistent implementation (load without iterating history)

File: blockchain/src/storage.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vldm) [reply blockchain/src/storage.rs]: But we need to decide in our project how to handle errors. 
For example, Rust team suggest, to panic if we not abble to recover:
https://doc.rust-lang.org/book/ch09-03-to-panic-or-not-to-panic.html

[the remark this answers is not in the document]
Comment (rtsisyk) [reply blockchain/src/storage.rs]: Column families are OK.

[the remark this answers is not in the document]
Comment (rtsisyk) [reply blockchain/src/storage.rs]: We are able to handle disk errors. See register_key_block() for example - it already can be atomic.

File: node/src/lib.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (rtsisyk) [reply node/src/lib.rs]: This function is massive copy-pasted from existing one. Try to refactor it.

**Comment by rtsisyk** [comment]
I see ` I am facilitator.` two times:

2019-02-04 10:52:47 INFO [stegos_txpool] I am facilitator.
2019-02-04 10:52:47 INFO [stegos_txpool] I am facilitator.

Why? It seems to be a bug.

**Pull request merged**